### ISSUE TITLE
Type of VAR1 in DS2 code

### ISSUE THREAD

**Issue by Pulpanj** [sassoftware/python-sasctl#58]
**Describe the issue**
I am trying to register sklearn Decision tree model (classification one) into Model Manager and run its scoring in MM. Generated DS2 code returns errors due to invalid types of VAR1 (returned variable). Generated python use  result = obj.predict(inputrun) while classification tree evaluation in MM will require result = obj.predict_proba(inputrun) to get target (level) probabilities. Are there any means to control generated wrapper code ?

**To Reproduce**
Steps or example code to reproduce the issue.
#python code used for registration of model
hmeq = pd.read_csv('/home/viyademo01/demodata/hmeq.csv',sep= ',')
target = hmeq['BAD']
# Specify the categorical target variable
y = target.astype('category')
# Retrieve the categories of the target variable
y_category = y.cat.categories
nYCat = len(y_category)
predictors = [ 'MORTDUE', 'VALUE', 'YOJ', 'DEROG', 'DELINQ', 'CLAGE', 'NINQ', 'CLNO', 'DEBTINC']
X = hmeq[predictors] # Features
y = hmeq.BAD # Target variable
# Split dataset into training set and test set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=1) # 70% training and 30% test
# X_train.fillna(X_train.mean(), inplace=True)
np.nan_to_num(X_train,copy=False,nan=0)
# Create Decision Tree classifer object
clf = DecisionTreeClassifier(max_leaf_nodes=10)
# Train Decision Tree Classifer
clf = clf.fit(X_train,y_train)
from sasctl import Session
from sasctl.tasks import register_model, publish_model, update_model_performance
from sasctl.services import model_repository as mr
from sasctl.services import model_management as mm
s=Session(host, username,password)
project = 'HMEQ Open DT12'
model_name = 'HMEQ Open DT12'
# Register the model in SAS Model Manager
reg=register_model(clf,
               model_name,
               input=X_train,       # Use X to determine model inputs
               project=project,     # Register in  project
               force=True,           # Create project if it doesn't exist
               version ='new'
            )         
# all goes well but DS2 code declares VAR1 as int ...

**Expected behavior**
Generated python use  result = obj.predict(inputrun) while classification tree evaluation in MM will require result = obj.predict_proba(inputrun) to get target (level) probabilities. Are there any means to control generated wrapper code ?

**Stack Trace**
If you're experiencing an exception, include the full stack trace and error message.

**Version**
What version of `sasctl` are you using?
1.5

**Comment by jlwalke2** [comment]
Closing this ticket as it has been addressed outside of GitHub, but I'll summarize the results here for posterity.

1. The `register_model` task attempts to generate working score code for SAS and Python models.

1. As of v1.5.3 it is not possible to customize the Python score code that's automatically generated.

1. However, it is possible to register a model with a custom set of metadata and files.  See the [register_custom_model](https://github.com/sassoftware/python-sasctl/blob/master/examples/register_custom_model.py) example for details.

Any model that does not neatly fit the interface expected by `register_model` can still be registered by creating any necessary custom wrappers or score code and then manually passing them in the `files=` parameter.
